src/neuroconv/tools/roiextractors/roiextractors.py

```diff
@@ -227,39 +230,9 @@ def add_devices_to_nwbfile(nwbfile: NWBFile, metadata: dict | None = None) -> NW
     return nwbfile
 
 
-def _create_imaging_plane_from_metadata(nwbfile: NWBFile, imaging_plane_metadata: dict) -> ImagingPlane:
-    """
-    Private auxiliary function to create an ImagingPlane object from pynwb using the imaging_plane_metadata.
-
-    Parameters
-    ----------
-    nwbfile : NWBFile
-        An previously defined -in memory- NWBFile.
-
-    imaging_plane_metadata : dict
-        The metadata to create the ImagingPlane object.
-
-    Returns
-    -------
-    ImagingPlane
-        The created ImagingPlane.
-    """
-
-    device_name = imaging_plane_metadata["device"]
-    imaging_plane_metadata["device"] = nwbfile.devices[device_name]
-
-    imaging_plane_metadata["optical_channel"] = [
-        OpticalChannel(**metadata) for metadata in imaging_plane_metadata["optical_channel"]
-    ]
-
-    imaging_plane = ImagingPlane(**imaging_plane_metadata)
-
-    return imaging_plane
-
-
 def add_imaging_plane_to_nwbfile(
     nwbfile: NWBFile,
-    metadata: dict,
+    metadata: dict | None = None,
```

Comment:
> Wait, actually, I don't think that metadata should ever be None at this level.

In that case, it should stay mandatory in the function signature and all the places where it's assumed that it could be None should be updated.

Reply:
But to be clear, metadata["Ophys"] could be an empty dictionary, right? And I don't think we are testing that case for this function.